### PR TITLE
Jest add flexibility

### PR DESCRIPTION
**Pull request by CodeByAlex**
## What I did
Refactored the Jest code a bit from being an MVP to a more sustainable codebase. Built-in flexibility for adding new types of test status' that will help @fabiradi work on his bugfix/feature outlined here: #7740. 
Sorted values coming into the status bar so that it will order the parts from smallest to largest
slightly improved performance by using fewer loops in render function

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: addons/jest/src/components/Panel.tsx [hunk 1]
```diff
@@ -58,34 +66,60 @@ const SuiteTotals = styled(({ successNumber, failedNumber, result, className, wi
...
 interface ContentProps {
   tests: Test[];
   className?: string;
 }
 
+const getTestsByTypeMap = (result: any) => {
+  const testsByType: Map<string, any> = new Map();
+  result.assertionResults.forEach((assertion: any) => {
+    // using switch to allow for new types to be added
+    switch (assertion.status) {
```
Comment (shilman) [inline addons/jest/src/components/Panel.tsx]: why not just use `assertion.status` instead of switch because the logic is the same in each branch?

**Review by shilman — APPROVED** [review]
LGTM with possible exception of duplicate logic!

**Comment by shilman** [comment]
Also, what's the bug?

File: addons/jest/src/components/Panel.tsx (hunk 1 above)
Comment (CodeByAlex) [reply addons/jest/src/components/Panel.tsx]: Oh my goodness, nice find. I could definitely use assertion.status instead. No need for the switch at all

**Comment by CodeByAlex** [comment]
@shilman switched it to maintenance. Do you think it would fall under that category?

**Comment by shilman** [comment]
@CodeByAlex that was my thought 👍

File: addons/jest/src/components/Panel.tsx (hunk 1 above)
Comment (CodeByAlex) [reply addons/jest/src/components/Panel.tsx]: Done!

**Pull request merged**